ambari-server/src/main/java/org/apache/ambari/server/controller/AmbariServer.java

```diff
@@ -633,6 +634,7 @@ private ServerConnector createSelectChannelConnectorForAgent(Server server, int
 
 
       HttpConfiguration https_config = new HttpConfiguration();
+      https_config.addCustomizer(new SecureRequestCustomizer());
```

Comment:
Does this correct the issue? I thought that the `SecureRequestCustomizer` will need to be configured to send or not send the header depending on the configuration of Ambari.

Reply:
Customizer is already removed for API server.
This PR returns it for agent server, as some properties configured by it seems to be required for agent communication.

Reply:
awesome.. thanks for the clarification.